packages/xo-server/src/api/vm.js

```diff
@@ -756,6 +757,30 @@ export { convertToTemplate as convert }
 
 // -------------------------------------------------------------------
 
+export async function cloneToTemplate({ vm }) {
+  if (vm.power_state === 'Running') {
+    throw incorrectState({
+      actual: 'Running',
+      expected: 'Halted',
+      object: vm.$id,
+      property: 'power_state',
+    })
+  }
+  const clonedVm = await this.getXapi(vm).copyVm(vm._xapiId, vm._xapiId, {
```

Comment:
But in our case, we don't know which SR have to be used. By sending one VM ID, the method is reusing the SR of the VM passed as second param.
Should i ask to the user in which SR he want to store this one ?

Reply:
Normally, `VM.copy` should reject if the passed ID doesn't correspond to an SR.

So, I suggest to make the SR ID optional in `copyVM` because it's optional is the low-level implementation.